### PR TITLE
chore(namespace): switch namespace to kube-system

### PR DESCRIPTION
**Pull request by charliedmcb**
Fixes # 

**Description**
This PR is a demo of the idea of switching the namespace from `karpenter` to `kube-system`

**How was this change tested?**

*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

File: hack/monitoring/prometheus-values.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ extraScrapeConfigs: |
       - role: endpoints
         namespaces:
           names:
-          - karpenter
+          - karpenter # Does this need to be updated?
```
Comment (Bryce-Soghigian) [inline hack/monitoring/prometheus-values.yaml:25]: yes

File: pkg/alt/karpenter-core/pkg/operator/operator.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ const (
 func NewOperator() (context.Context, *coreoperator.Operator) {
 	overlayNamespace := os.Getenv("OVERLAY_NAMESPACE")
 	if overlayNamespace == "" {
-		overlayNamespace = "karpenter-system"
+		overlayNamespace = "karpenter-system" // Note: this namespace goes counter to our namespace usagage elsewhere
```
Comment (Bryce-Soghigian) [inline pkg/alt/karpenter-core/pkg/operator/operator.go:85]:
```suggestion
		overlayNamespace = "karpenter-system" // Note: this namespace goes counter to our namespace usage elsewhere
```

File: pkg/alt/karpenter-core/pkg/operator/operator.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ const (
 func NewOperator() (context.Context, *coreoperator.Operator) {
 	overlayNamespace := os.Getenv("OVERLAY_NAMESPACE")
 	if overlayNamespace == "" {
-		overlayNamespace = "karpenter-system"
```
Comment (Bryce-Soghigian) [inline pkg/alt/karpenter-core/pkg/operator/operator.go:85]: where do we set `OVERLAY_NAMESPACE`

File: pkg/metrics/constants.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ package metrics
 
 const (
 	// Namespace(s).
-	Namespace = "karpenter"
+	Namespace = "kube-system"
```
Comment (Bryce-Soghigian) [inline pkg/metrics/constants.go:21]: We don't want ot change this. This is the prometheus concept of namespace not kubernetes. 

See: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Opts.Namespace 


it is used as the prefix to our metrics

File: test/pkg/debug/events.go [hunk 5]
```diff
@@ -52,7 +52,7 @@ func (c *EventClient) DumpEvents(ctx context.Context) error {
 
 func (c *EventClient) dumpKarpenterEvents(ctx context.Context) error {
 	el := &v1.EventList{}
-	if err := c.kubeClient.List(ctx, el, client.InNamespace("karpenter")); err != nil {
+	if err := c.kubeClient.List(ctx, el, client.InNamespace("kube-system")); err != nil {
```
Comment (Bryce-Soghigian) [inline test/pkg/debug/events.go:55]: Feels like this should be a constant somewhere rather than us typing it in manually

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

**Comment by tallaxes** [comment]
This got absorbed by #528 (@charliedmcb, please take a look)

**Pull request closed without merging**